helios-services/src/main/java/com/spotify/helios/master/ZooKeeperMasterModel.java

```diff
@@ -593,8 +593,13 @@ public void rollingUpdateStep(final DeploymentGroup deploymentGroup,
           zkOperations.toArray(new ZooKeeperOperation[zkOperations.size()])
       ));
     } catch (final KeeperException e) {
-      throw new HeliosRuntimeException(
-          "rolling-update on deployment-group " + deploymentGroup.getName() + " failed", e);
+      if ((e instanceof KeeperException.BadVersionException) && e.getPath().equals(statusPath)) {
```

Comment:
@rohansingh I just saw in the master logs that `e.getPath()` sometimes throws NPE because e.path is null. What should we do in that case?